var/spack/repos/builtin/packages/aws-parallelcluster/package.py

```diff
@@ -33,13 +34,17 @@ class AwsParallelcluster(PythonPackage):
 
     depends_on('python@2.7:', type=('build', 'run'))
     depends_on('py-setuptools', type=('build', 'run'))
-    depends_on('py-boto3@1.10.15:', type=('build', 'run'))
-    depends_on('py-future@0.16.0:0.18.2', type=('build', 'run'))
-    depends_on('py-tabulate@0.8.2:0.8.3', type=('build', 'run'))
+    depends_on('py-boto3@1.10.15:', when='@:2.8', type=('build', 'run'))
+    depends_on('py-boto3@1.14.3:', when='@2.9:', type=('build', 'run'))
+    depends_on('py-future@0.16.0:0.18.2', when='@:2.8', type=('build', 'run'))
+    depends_on('py-future@0.18.2:', when='@2.9:', type=('build', 'run'))
+    depends_on('py-tabulate@0.8.2:0.8.7', type=('build', 'run'))
     depends_on('py-ipaddress@1.0.22:', type=('build', 'run'))
     depends_on('py-enum34@1.1.6:', when='^python@:3.3', type=('build', 'run'))
-    depends_on('py-pyyaml@5.1.2:', type=('build', 'run'))
+    depends_on('py-pyyaml@5.1.2:', when='@:2.8', type=('build', 'run'))
+    depends_on('py-pyyaml@5.3.1:', when='@2.9:', type=('build', 'run'))
     depends_on('py-configparser@3.5.0:3.8.1', when='^python@:2', type=('build', 'run'))
+    depends_on('py-jinja2@2.11.0:', when='@2.9:', type=('build', 'run'))
```

Comment:
Many of these have specific version requirements for different Python versions. This can be represented as:
```python
depends_on('py-jinja2@2.10.1', when='@2.9: ^python@3.4.0:3.4.999', type=('build', 'run'))
depends_on('py-jinja2@2.11.0:', when='@2.9: ^python@:3.3,3.5:', type=('build', 'run'))
```